internal/telemetry/telemetrytest/telemetrytest.go

```diff
@@ -21,27 +21,41 @@ type MockClient struct {
 	AsmEnabled      bool
 }
 
-// Start starts and adds configuration data to the mock client.
-func (c *MockClient) Start(configuration []telemetry.Configuration) {
+// ProductStart starts and adds configuration data to the mock client.
+func (c *MockClient) ProductStart(namespace telemetry.Namespace, configuration []telemetry.Configuration) {
```

Comment:
It'd be preferable to mock out as little of the real code as possible. For example, we can just sub out the `productChange` function that's called in `ProductStart` for the tests where you don't want to real `productChange` function, but use the rest of the real code in `ProductStart`. Same with subbing out `c.start`. We can chat offline about this if you'd like

Reply:
*Talked offline